google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITQueryAggregationsTest.java

```diff
@@ -1,5 +1,750 @@
+    assertThat(snapshot.get(sum("rating"))).isEqualTo(Double.NaN);
+  }
+
+  @Test
+  public void performsSumOverResultSetOfZeroDocuments() throws Exception {
+    CollectionReference collection = testCollectionWithDocs(testDocs1);
+    AggregateQuerySnapshot snapshot =
+        collection.whereGreaterThan("pages", 200).aggregate(sum("pages")).get().get();
+    assertThat(snapshot.get(sum("pages"))).isEqualTo(0);
+  }
+
+  @Test
+  public void performsSumOnlyOnNumericFields() throws Exception {
+    Map<String, Map<String, Object>> testDocs =
+        map(
+            "a", map("author", "authorA", "title", "titleA", "rating", 5),
+            "b", map("author", "authorB", "title", "titleB", "rating", 4),
+            "c", map("author", "authorC", "title", "titleC", "rating", "3"),
+            "d", map("author", "authorD", "title", "titleD", "rating", 1));
+    CollectionReference collection = testCollectionWithDocs(testDocs);
+    AggregateQuerySnapshot snapshot =
+        collection.aggregate(sum("rating"), AggregateField.count()).get().get();
+    assertThat(snapshot.get(sum("rating"))).isEqualTo(10);
+    assertThat(snapshot.get(AggregateField.count())).isEqualTo(4);
+  }
+
+  @Test
+  public void performsSumOfMinIEEE754() throws Exception {
+    Map<String, Map<String, Object>> testDocs =
+        map("a", map("author", "authorA", "title", "titleA", "rating", Double.MIN_VALUE));
+    CollectionReference collection = testCollectionWithDocs(testDocs);
+    AggregateQuerySnapshot snapshot = collection.aggregate(sum("rating")).get().get();
+    assertThat(snapshot.get(sum("rating"))).isEqualTo(Double.MIN_VALUE);
+  }
+
+  @Test
+  public void performsAverageOfIntsThatResultsInAnInt() throws Exception {
+    Map<String, Map<String, Object>> testDocs =
+        map(
+            "a", map("author", "authorA", "title", "titleA", "rating", 10),
+            "b", map("author", "authorB", "title", "titleB", "rating", 5),
+            "c", map("author", "authorC", "title", "titleC", "rating", 0));
+    CollectionReference collection = testCollectionWithDocs(testDocs);
+    AggregateQuerySnapshot snapshot = collection.aggregate(average("rating")).get().get();
+    assertThat(snapshot.get(average("rating"))).isEqualTo(5);
+    assertThat(snapshot.getLong(average("rating"))).isEqualTo(5L);
+    assertThat(snapshot.getDouble(average("rating"))).isEqualTo(5.0);
+  }
+
+  @Test
+  public void performsAverageOfFloatsThatResultsInAnInt() throws Exception {
+    Map<String, Map<String, Object>> testDocs =
+        map(
+            "a", map("author", "authorA", "title", "titleA", "rating", 10.5),
+            "b", map("author", "authorB", "title", "titleB", "rating", 9.5));
+    CollectionReference collection = testCollectionWithDocs(testDocs);
+    AggregateQuerySnapshot snapshot = collection.aggregate(average("rating")).get().get();
+    assertThat(snapshot.get(average("rating"))).isEqualTo(10);
```

Comment:
Consider asserting that the type is actually Double, because average is "always Double"

Reply:
Done